avro/avro.go

```diff
@@ -461,3 +510,16 @@ func unmarshalStruct(val interface{}, v reflect.Value, i int) (interface{}, erro
 
 	return item.Interface(), nil
 }
+
+func unmarshalMap(value interface{}) map[string]string {
+	return unmarshalStringMap(value)
```

Comment:
`unmarshalMap` should handle value of nil and return nil, like so

```
if value == nil {
  return nil
}
```

Reply:
Extend unit tests for empty map

Reply:
I'll look at this